rfcs/proposed/task_group_dynamic_dependencies/README.md

```diff
@@ -0,0 +1,365 @@
+our task_group extension to cover.
+
+The key capability required for recursive decomposition is the ability to 
+create work while executing a task and insert this newly created work before 
+the (perhaps already submitted) successors of the currently executing task. 
+As a simple example, consider a merge sort. As shown in the figure that 
+follows, the top-level algorithm breaks a collection into two pieces and 
+creates three tasks:
+
+1. a task to sort the left half
+2. a task to sort the right half
+3. a task to merge the left and right sorted halves once they have been sorted. 
+
+In a recursive merge sort, each of the sort tasks recursively takes the same 
+approach to sort their portions of the collection. The top-level task (and 
+subsequent recursively generated tasks) must be able to create new tasks 
+and then update the graph so that their outer merge task waits for the 
+newly created subtasks to complete.
+
+<img src="merge_sort.png" width=800>
+
+A key point about this recursive parallel algorithm is that we must change 
+the predecessors of the merge tasks. But the merge tasks are already 
+submitted at the time their predecessors are modified! In the previous 
+section, we noted that updating the predecessors of a submitted task is 
+risky, because there is a potential race.  However, in the example shown 
+here, we know it’s safe to add additional predecessors to the merge task 
+because it simply cannot start executing until all its current predecessors 
+complete, and its predecessors are the tasks modifying the predecessors! 
+
+We therefore propose a very limited extension that allows the transfer of 
+all the successors of a currently executing task to become the successors 
+of a different created task. This function can only access the successors
+of the currently executing task, and those tasks are prevented from executing 
+by a dependence on the current task itself, so we can ensure that we can safely 
+update the incoming dependencies for those tasks without worrying about any 
+potential race.
+
+One possible spelling for this function would be `transfer_successors_to(h)`, 
+where `h` is a `task_handle` to a created task and the 
+`transfer_successors_to` function must be called from within a task. Calling
+this function from outside a task, or passing anything other than a `task_handle`
+representing a task in the created state is undefined behavior.
+
+### Proposed changes to task_handle
+
+    namespace oneapi {
+    namespace tbb {
+        class task_handle {
+        public:
+
+            // existing functions
+            task_handle();
+            task_handle(task_handle&& src);
+            ~task_handle();
+            task_handle& operator=(task_handle&& th);
+            explicit operator bool() const noexcept;
+
+            // proposed additions
+            void add_predecessor(task_handle& th);
+            void add_successor(task_handle& th);
```

Comment:
If we speak about the minimally required API, I doubt we need two variants of the same function, just with different name and the order of arguments. I would only keep one.